src/D2L.CodeStyle.Analyzers/Immutability/ReadOnlyParameterAnalyzer.cs

```diff
@@ -73,7 +73,13 @@ ParameterSyntax syntax
 			}
 
 			IMethodSymbol method = parameter.ContainingSymbol as IMethodSymbol;
-			BlockSyntax methodBody = ( method.DeclaringSyntaxReferences[0].GetSyntax( ctx.CancellationToken ) as BaseMethodDeclarationSyntax ).Body;
+			SyntaxNode methodSyntaxNode = method.DeclaringSyntaxReferences[ 0 ].GetSyntax( ctx.CancellationToken );

```

Comment:
There's probably also something with partial methods, and this should actually find the DeclaringSyntaxReference which _has_ a body.